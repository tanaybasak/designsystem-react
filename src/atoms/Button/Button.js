import React from 'react';
import PropTypes from 'prop-types';
import prefix from '../../settings';

const Button = ({
  type,
  small,
  className,
  children,
  onClick,
  ...restProps
}) => {
  const classNameOptions = {
    primary: 'hcl-primary',
    'primary-danger': 'hcl-primary hcl-danger',
    secondary: 'hcl-secondary',
    'secondary-danger': 'hcl-secondary hcl-danger',
    ghost: 'hcl-ghost'
  };

  const classNameType = classNameOptions[type] ? classNameOptions[type] : '';
  const smallBtn = small ? 'hcl-sm' : '';
  const classnames = `${prefix}-btn ${className} ${classNameType} ${smallBtn}`.trim();

  return (
    <button
      type="button"
      className={classnames}
      {...restProps}
      onClick={onClick}
    >
      {children}
    </button>
  );
};

Button.propTypes = {
  /** button content */
  children: PropTypes.any,
  /** type of button eg : primary , primary-danger , secondary-danger , secondary, ghost  */
  type: PropTypes.string,
  /** Style class of the component */
  className: PropTypes.string,
  /** small button  */
  small: PropTypes.bool,
  /** call back function on click  */
  onClick: PropTypes.func
};

Button.defaultProps = {
  children: 'Click Me',
  className: '',
  type: '',
  small: false,
  disabled: false,
  onClick: () => {}
};

export default Button;
