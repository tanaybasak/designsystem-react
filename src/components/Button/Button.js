import React from 'react';
import PropTypes from "prop-types";
import { prefix } from '../../settings';

const Button = React.forwardRef((props, ref) => (
    <button
        ref={ref}
        className={`${prefix}-btn ${prefix}-btn--${props.className}`}
        {...props.data}
        onClick={props.onClick}
    >
        {props.label}
    </button>
));

Button.propTypes = {
    label: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
    className: PropTypes.string.isRequired,
    data: PropTypes.object
};

export default Button;