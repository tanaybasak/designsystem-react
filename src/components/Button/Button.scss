$primaryBlue: #0578c3;
$primaryBlueHover: #0066b3;
$white: #ffffff;
$disabled: #aaaaaa;
$dark: #000000;
$darkHover: #4b4b4b;

@mixin button($bgColor, $textColor, $borderColor, $hoverBgColor, $hoverColor, $hoverBorderColor) {
    background-color: $bgColor;
    border: 1px solid $borderColor;
    color: $textColor;
    &:hover {
        background-color: $hoverBgColor;
        color: $hoverColor;
        border: 1px solid $hoverBorderColor;
    }
}

.patron-btn {
    min-height: 35px;
    outline: none;
    &.patron-btn-primary {
        @include button($primaryBlue, $white, $primaryBlue, $primaryBlueHover, $white, $primaryBlue);
    }
    &.patron-btn-outline-primary {
        @include button($white, $primaryBlue, $primaryBlue, $primaryBlueHover, $white, $primaryBlue);
    }

    &.patron-btn-dark {
        @include button($dark, $white, $dark, $darkHover, $white, $darkHover);
    }
    &.patron-btn-outline-dark {
        @include button($white, $darkHover, $darkHover, $darkHover, $white, $darkHover);
    }

    &.patron-btn-info {}
    &[disabled] {
        @include button($disabled, $white, $disabled, $disabled, $white, $disabled);
    }
}