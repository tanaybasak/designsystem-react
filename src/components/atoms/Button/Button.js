import React from 'react';
import PropTypes from 'prop-types';
import prefix from '../../../settings';

const Button = ({ className, children, onClick, ...restProps }) => {
  const classnames = `${prefix}-btn ${className}`.trim();

  return (
    <button
      type='button'
      className={classnames}
      {...restProps}
      onClick={onClick}
    >
      {children}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.string,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  onClick: PropTypes.func
};

Button.defaultProps = {
  children: 'Click Me',
  className: '',
  disabled: false,
  onClick: () => {}
};

export default Button;
