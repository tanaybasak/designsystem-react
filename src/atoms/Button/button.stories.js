import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { select, text, boolean } from '@storybook/addon-knobs';
//@update-path-build-start
import Button from './Button';
//@update-path-build-end
import icons from '../../../.storybook/iconList';

const classOptions = [
  'primary',
  'primary-danger',
  'secondary',
  'secondary-danger',
  'ghost',
  'warning',
  'neutral'
];

const typeOptions = ['button', 'reset', 'submit'];

storiesOf('Components/Button', module)
  .add(
    'default',
    () => (
      <Button
        type={select('Type', classOptions, 'primary')}
        kind={select('Kind', typeOptions, 'button')}
        title="Default"
        disabled={boolean('Disabled', false)}
        small={boolean('Small', false)}
      >
        {text('Label', 'Click Me')}
      </Button>
    ),
    {
      info: {
        text: `Description About Button Component`,
        document: ['Button']
      }
    }
  )
  .add(
    'with icon',
    () => (
      <Button
        type={select('Type', classOptions, 'primary')}
        kind={select('Kind', typeOptions, 'button')}
        title="Default"
        disabled={boolean('Disabled', false)}
        small={boolean('Small', false)}
      >
        <i
          className={`p-hclsw p-hclsw-${select('Icon Class', icons, 'user')}`}
        />
      </Button>
    ),
    {
      info: {
        text: `Description About Button Component 
      `,
        document: ['Button']
      }
    }
  )
  .add(
    'with text and icon',
    () => (
      <Button
        type={select('Type', classOptions, 'primary')}
        kind={select('Kind', typeOptions, 'button')}
        title="Default"
        disabled={boolean('Disabled', false)}
        small={boolean('Small', false)}
      >
        {text('Label', 'Click Me')}
        <i
          className={`hcl-btn-icon p-hclsw p-hclsw-${select(
            'Icon Class',
            icons,
            'user'
          )}`}
        />
      </Button>
    ),
    {
      info: {
        text: `Description About Button Component`,
        document: ['Button']
      }
    }
  );
