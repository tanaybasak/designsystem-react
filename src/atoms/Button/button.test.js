import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Button from './Button';

describe('<Button> component', () => {
    it('snapshots/renders Button correctly', () => {
        const tree = renderer.create(<Button />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('simulate click event in Button and is called twice', () => {
        const mockCallBack = jest.fn();
        const wrapper = shallow((<Button onClick={mockCallBack}>Click</Button>));
        wrapper.find('button').simulate('click');
        wrapper.find('button').simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(2);
    });
});