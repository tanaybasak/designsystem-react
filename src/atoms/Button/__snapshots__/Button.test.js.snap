// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> component snapshots/renders Button correctly 1`] = `
<button
  className="hcl-btn  hcl-primary"
  disabled={false}
  type="button"
>
  Click Me
</button>
`;
